WordPress/Classes/Uploads/UploadsManager.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import Reachability
+
+/// Takes care of coordinating all uploaders used by the app.
+///
+class UploadsManager: NSObject {
+    private let uploaders: [Uploader]
+    private let reachability: Reachability = Reachability.forInternetConnection()
+
+    // MARK: Initialization & Finalization
+
+    /// Default initializer.
+    ///
+    /// - Parameters:
+    ///     - uploaders: the uploaders that this object will be handling.
+    ///
+    required init(uploaders: [Uploader]) {
+        self.uploaders = uploaders
+
+        super.init()
+
+        setupReachableBlock()
+    }
+
+    // MARK: Reachability
+
+    private func setupReachableBlock() {
+        reachability.reachableBlock = { _ in
```

Comment:
Two questions here:
1. Shouldn't this be `{ [weak self] in
self?.resume()
}

2. Shouldn't we check if reachability isReachable() and resume if it is?
like  `reachability.reachableBlock = { [weak self] reachability in
guard let reachability = reachability else {
return
}
if reachability.isReachable() {
}`

Reply:
1. Yep.  Will change.
2. Not really because in theory `reachableBlock` is only called when a connection is detected (as opposed to `unreachableBlock`).

Reply:
Fixed 1.  There should be no need to fix 2 as described above.